lib/matplotlib/axes/_base.py

```diff
@@ -169,25 +169,25 @@ def _process_plot_format(fmt):
         if fmt[i:i+2] in mlines.lineStyles:  # First, the two-char styles.
             if linestyle is not None:
                 raise ValueError(
-                    'Illegal format string "%s"; two linestyle symbols' % fmt)
+                    f'Illegal format string {fmt!r}; two linestyle symbols')
             linestyle = fmt[i:i+2]
             i += 2
         elif c in mlines.lineStyles:
             if linestyle is not None:
                 raise ValueError(
-                    'Illegal format string "%s"; two linestyle symbols' % fmt)
+                    f'Illegal format string {fmt!r}; two linestyle symbols')
```

Comment:
I doubt that this can be triggered. Something like `plt.plot((1, 1), '-', linestyle='solid')` gives
```
UserWarning: linestyle is redundantly defined by the 'linestyle' keyword argument and the fmt string "-" (-> linestyle='-'). The keyword argument will take precedence.
```